### ISSUE TITLE
SvgDOMManager:  ID lookup fails on build()

### ISSUE THREAD

**Issue by jimmyff** [zathras/jovial_svg#113]
![Screenshot 2024-09-05 at 11 12 08](https://github.com/user-attachments/assets/ee908b02-ff24-46fd-82ed-a0e852b7f31b)

Left is loaded via `ScalableImage.fromSvgString(...)` and right has been loaded  through `SvgDOMManager`:

```dart
final svg = SvgDOMManager.fromString(svgSource);
setState(() {
_si = svg.build();
});
```

SVG file attached:
![jovial_bug](https://github.com/user-attachments/assets/8b60528c-766b-40b3-ba31-103c1ace7509)

**Comment by zathras** [comment]
Please include a complete, minimal program that reproduces the problem.

**Comment by zathras** [comment]
Often, when a complete example to reproduce a problem is lacking, it turns out to be a problem outside the library.  But it this case, I was able to reproduce and narrow down the problem pretty quickly.  It turns out to have been a problem looking up node IDs in a cloned SvgDOM instance.

**Comment by jimmyff** [comment]
Thanks @zathras I'll fork and test now, also I asked a few best practice related questions over in discussions, don't suppose if you have a moment you could see if you can comment on any?

Thanks again, this is a gem of a dart package

**Comment by jimmyff** [comment]
Tested and my mermaid has her eyes, thanks! 👍

**Comment by zathras** [comment]
I like to leave issues open, and marked "fixed in repo" until I actually publish the fix.

**Comment by zathras** [comment]
Published in 1.1.23
